src/eureka/S5_lightcurve_fitting/models/FleckModel.py

```diff
@@ -0,0 +1,245 @@
+                value = self.parameters.dict[item0][0]
+                spotcon[chan] = value
+                # read stellar inclination
+                item0 = 'spotstari' + pid_id
+                if self.parameters.dict[item0][1] == 'free':
+                    item0 += channel_id
+                value = self.parameters.dict[item0][0]
+                starinc[chan] = value
+                # read number of points
+                npoints = self.parameters.dict['spotnpts'][0]
+                # read stellar rotation (if provided)
+                if 'spotrot' in self.parameters.dict.keys():
+                    starrot[chan] = self.parameters.dict['spotrot'][0]
+                    fleck_fast = False
+                            
+                # Set limb darkening parameters
+                uarray = []
+                for u in self.coeffs:
+                    index = np.where(np.array(self.paramtitles) == u)[0]
+                    if len(index) != 0:
+                        item = self.longparamlist[chan][index[0]]
+                        uarray.append(self.parameters.dict[item][0])
+                pl_params.u = uarray
+                pl_params.limb_dark = self.parameters.dict['limb_dark'][0]
+
+                # Enforce physicality to avoid crashes from batman by returning
+                # something that should be a horrible fit
+                if not ((0 < pl_params.per) and (0 < pl_params.inc < 90) and
+                        (1 < pl_params.a) and (0 <= pl_params.ecc < 1)):
+                    # Returning nans or infs breaks the fits, so this was the
+                    # best I could think of
+                    light_curve = 1e12*np.ma.ones(time.shape)
+                    continue
+
+                # Use batman ld_profile name
+                if self.parameters.limb_dark.value == '4-parameter':
+                    pl_params.limb_dark = 'nonlinear'
+                elif self.parameters.limb_dark.value == 'kipping2013':
+                    # Enforce physicality to avoid crashes from batman by
+                    # returning something that should be a horrible fit
+                    if pl_params.u[0] <= 0:
+                        # Returning nans or infs breaks the fits, so this was
+                        # the best I could think of
+                        lcfinal = 1e12*np.ma.ones(time.shape)
+                        continue
+                    pl_params.limb_dark = 'quadratic'
+                    u1 = 2*np.sqrt(pl_params.u[0])*pl_params.u[1]
+                    u2 = np.sqrt(pl_params.u[0])*(1-2*pl_params.u[1])
+                    pl_params.u = np.array([u1, u2])
+
+                # Make the star object
+                star = fleck.Star(spot_contrast=spotcon[chan], 
+                                  u_ld=pl_params.u, 
+                                  rotation_period=starrot[chan])
+                # Make the transit model
+                fleck_times = np.linspace(time[0], time[-1], npoints)
```

Comment:
This will fail if the first or last time values are masked (which does sometimes happen)

Reply:
ah yes, at some point I had changed this to be the unmasked array and apparently that got lost along the way. Will fix